### PR TITLE
build(deps): bump react-native from 0.72.12 to 0.72.17

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [] 

### Description

Bumps react-native from 0.72.12 to 0.72.17 till we manage to fully upgrade to 0.73.

Haven't thoroughly test it yet but I did a sanity check that the app is working.

We can merge and test it out till next release.

- Also bumped up the react native patch - hope we can get rid of it at some point in the future.

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- bump react-native from 0.72.12 to 0.72.17 - gkartalis

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: ios/Artsy/App_Resources/PrivacyInfo.xcprivacy [hunk 1]
```diff
@@ -2,158 +2,163 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
-	<key>NSPrivacyTracking</key>
-	<false/>
-	<key>NSPrivacyTrackingDomains</key>
-	<array/>
+	<key>NSPrivacyAccessedAPITypes</key>
```
Comment (gkartalis) [inline ios/Artsy/App_Resources/PrivacyInfo.xcprivacy:5]: not that important changes, after building XCode decided to shuffle things up so I will leave it as is 🤷🤷🤷🤷

LMK if you have any concerns about this

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (bump react-native from 0.72.12 to 0.72.17 - gkartalis)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against b405a0af2cce9e3329caf3e2378504fc59283920
</p>

**Review by MrSltun — APPROVED** [review]

**Comment by gkartalis** [comment]
Aaah need to update the pod cache thats why it failed

**Review by araujobarret — DISMISSED** [review]
💪

**Comment by gkartalis** [comment]
before we merge this we will need to send a message to #practice-mobile for folks to run `yarn install:all &&  bundle exec pod update hermes-engine --no-repo-update` when they get the latest main.

**Pull request merged**